File: uk-ets-app-registry/uk-ets-registry-api/src/main/java/gov/uk/ets/registry/api/account/web/model/AccountStatusAction.java
package gov.uk.ets.registry.api.account.web.model;

import lombok.Getter;

/**
 * Possible account status change actions
 * 
 * @author fragkise
 */
@Getter
public enum AccountStatusAction {

	RESTRICT_SOME_TRANSACTIONS("Restrict some transactions",
	                           "This will not restrict Surrender, Reverse Allocation or " +
	                           "Return Excess Allocation transactions for the account."),
	REMOVE_RESTRICTIONS("Remove restrictions", "Allow unrestricted transactions for the account."),
	SUSPEND_PARTIALLY("Suspend account (partially)", "Authorised Representatives will not be able " +
                                "to access or view the account. The account will still be able to receive units."),
	SUSPEND("Suspend account (fully)", "Authorised Representatives will not be able " +
	                                   "to access or view the account. The account will not be able to receive units."),
	UNSUSPEND("Unsuspend account", "Authorised Representatives will be able to access the account. " +
	                               "The account will be able to receive units.");

	private final String description;
	private final String hint;

	AccountStatusAction(String description, String hint) {
		this.description = description;
		this.hint = hint;
	}
}

File: uk-ets-app-registry/uk-ets-registry-web/src/app/shared/model/account/account-status-action.ts
import { AccountStatus } from '.';

export type AccountStatusAction =
  | 'RESTRICT_SOME_TRANSACTIONS'
  | 'REMOVE_RESTRICTIONS'
  | 'SUSPEND_PARTIALLY'
  | 'SUSPEND'
  | 'UNSUSPEND';

export interface AccountStatusActionOption {
  label: string;
  hint: string;
  value: AccountStatusAction;
  enabled: true;
  newStatus: AccountStatus;
  message?: string;
}

export interface AccountStatusRequest {
  accountId: string;
  status: AccountStatus;
  comment: string;
}

export type AccountStatusActionState = Omit<
  AccountStatusActionOption,
  'enabled'
>;
